data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -81,12 +79,116 @@ static String post(String url, String payload, InputStream stream, Integer timeo
                     throw new DataServiceException(url, message, ex);
                 }
             }
-        } catch (JSONException | IOException | URISyntaxException e) {
+        } catch (JSONException | IOException e) {
             throw new DataClientException(url, "Error in post request:" + e.getMessage(), e);
         }
         return null;
     }
 
+    static InputStream postToStreamingOutput(String url, String payload, long timeoutMs, Map<String, String> headers) throws DataServiceException, DataClientException {
```

Comment:
Note that I intentionally don't allow the query to be streamed in via InputStream. It also doesn't make sense to offer a "leaveOpen" flag for the response's InputStream.